src/components/ToDoForm.vue

```diff
@@ -1,23 +1,127 @@
+        class="ml-6 flex flex-row-reverse phone:ml-4"
+      />
+    </div>
+    <textarea
+      type="text"
+      class="ml-6 mt-6 flex h-24 w-10/12 text-left indent-1 font-header text-2xl font-semibold text-black phone:ml-4 phone:h-20 phone:w-56 phone:text-sm phone:font-normal phone:leading-4 phone:text-neutral-500"
+      :placeholder="updatedToDo.description"
+      v-model="updatedToDo.description"
+    ></textarea>
+    <div class="flex items-center justify-start">
+      <button
+        @click.prevent="saveTodo"
+        class="mb-4 ml-6 h-12 w-28 rounded-2xl bg-green-400 font-header text-lg font-semibold leading-6 text-white phone:ml-4 phone:h-8 phone:w-16 phone:rounded-lg phone:px-3 phone:py-1.5 phone:text-sm phone:leading-4"
+      >
+        Save
+      </button>
+      <button
+        @click.prevent="toggleDeletePopup"
+        class="mb-4 ml-4 h-12 w-28 rounded-2xl bg-neutral-200 font-header text-lg font-semibold leading-6 text-black phone:ml-4 phone:h-8 phone:w-16 phone:rounded-lg phone:px-3 phone:py-1.5 phone:text-sm phone:leading-4"
+      >
+        Delete
+      </button>
+    </div>
+    <DeletePopup
+      v-if="isDeletePopupVisible"
+      @deleteToDo="deleteToDo"
+      @cancelDelete="toggleDeletePopup"
+      ref="hideDeletePopupRef"
+    />
+  </div>
 </template>
 
 <script setup lang="ts">
 import { ref } from 'vue'
+import { onClickOutside } from '@vueuse/core'
+import ToDoPriority from './ToDoPriority.vue'
+import DeletePopup from './DeletePopup.vue'
 import { Todo } from '../todo.ts'
 
-const emit = defineEmits(['addToDo'])
+interface Props {
+  todo: Todo
+  index: number
+}
+
+const props = defineProps<Props>()
+
+const emit = defineEmits(['addToDo', 'deleteToDo', 'editToDo'])
 
-const todo = ref<Todo>({
-  title: 'Title',
-  description: 'blahblah',
-  priority: 'Medium',
-  isChecked: false
+const updatedToDo = ref<Todo>({
+  title: props.todo.title,
+  description: props.todo.description,
+  priority: props.todo.priority,
+  isChecked: props.todo.isChecked,
+  dueDate: props.todo.dueDate,
 })
 
-function handleClick() {
-  if (todo.value.title.length) {
-    emit('addToDo', todo.value)
+const isDropdownShowing = ref(false)
+const isDeletePopupVisible = ref(false)
+const hideDeletePopupRef = ref(null)
+const toDoDate = ref(formatDate())
+
+function formatDate() {
+  const date = new Date(updatedToDo.value.dueDate)
+  const year = date.getFullYear()
+  const month = date.getMonth() < 9 ? `0${date.getMonth() + 1}` : date.getMonth() + 1
+  const day = date.getDate() < 10 ? `0${date.getDate()}` : date.getDate()
+  return `${year}-${month}-${day}`
+}
+
+function updateNewPriority(priority: string) {
+  updatedToDo.value.priority = priority
+}
+
+function setDropdownVisibility(isShowing: boolean) {
+  isDropdownShowing.value = isShowing
+}
+
+function toggleDeletePopup() {
+  isDeletePopupVisible.value = !isDeletePopupVisible.value
+}
+
+function saveTodo() {
+  updatedToDo.value.dueDate = new Date(toDoDate.value)
```

Comment:
This should be moved after the validation which checks if the todoTitle is empty